src/integrationTest/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/resource/CreateGoalTest.kt

```diff
@@ -0,0 +1,128 @@
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.resource.model.ErrorResponse
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.resource.model.aValidCreateGoalRequest
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.resource.model.assertThat
+
+class CreateGoalTest : IntegrationTestBase() {
+
+  companion object {
+    private const val URI_TEMPLATE = "/action-plans/{prisonNumber}/goals"
+  }
+
+  @Test
+  fun `should return unauthorized given no bearer token`() {
+    webTestClient.post()
+      .uri(URI_TEMPLATE, aValidPrisonNumber())
+      .contentType(APPLICATION_JSON)
+      .exchange()
+      .expectStatus()
+      .isUnauthorized
+  }
+
+  @Test
+  fun `should return forbidden given bearer token with view only role`() {
+    webTestClient.post()
+      .uri(URI_TEMPLATE, aValidPrisonNumber())
+      .body(Mono.just(aValidCreateGoalRequest()), CreateGoalRequest::class.java)
+      .bearerToken(aValidTokenWithViewAuthority(privateKey = keyPair.private))
+      .contentType(APPLICATION_JSON)
+      .exchange()
+      .expectStatus()
+      .isForbidden
+  }
+
+  @Test
+  fun `should fail to create goal given no steps provided`() {
+    val prisonNumber = aValidPrisonNumber()
+    val createRequest = aValidCreateGoalRequest(steps = emptyList())
+
+    // When
+    val response = webTestClient.post()
+      .uri(URI_TEMPLATE, prisonNumber)
+      .body(Mono.just(createRequest), CreateGoalRequest::class.java)
+      .bearerToken(aValidTokenWithEditAuthority(privateKey = keyPair.private))
+      .contentType(APPLICATION_JSON)
+      .exchange()
+      .expectStatus()
+      .isBadRequest
+      .returnResult(ErrorResponse::class.java)
+
+    // Then
+    val actual = response.responseBody.blockFirst()
+    assertThat(actual)
+      .hasStatus(BAD_REQUEST.value())
+      .hasUserMessageContaining("At least one Step is required.")
+  }
+
+  @Test
+  @Transactional
+  fun `should add goal and to a new action plan given prisoner does not have an action plan`() {
+    // Given
+    val prisonNumber = aValidPrisonNumber()
+    val createRequest = aValidCreateGoalRequest()
+
+    // When
+    webTestClient.post()
+      .uri(URI_TEMPLATE, prisonNumber)
+      .body(Mono.just(createRequest), CreateGoalRequest::class.java)
+      .bearerToken(aValidTokenWithEditAuthority(privateKey = keyPair.private))
+      .contentType(APPLICATION_JSON)
+      .exchange()
+      .expectStatus()
+      .isCreated()
+
+    // Then
+    val actionPlan = actionPlanRepository.findByPrisonNumber(prisonNumber)
+    assertThat(actionPlan).isForPrisonNumber(prisonNumber)
+    assertThat(actionPlan).hasNumberOfGoals(1)
+    val goal = actionPlan!!.goals!![0]
+    assertThat(goal).hasTitle(createRequest.title)
+    assertThat(goal).hasReviewDate(createRequest.reviewDate)
+    assertThat(goal).hasNumberOfSteps(createRequest.steps.size)
+  }
+
+  @Test
+  @Transactional
+  fun `should add goal to prisoner's existing action plan`() {
```

Comment:
Hmmm ... this is kinda like an edit , rather than an add?
I guess its about context .... we are adding a goal ✅ , but editing an action plan ... 🤔